tests/run.go

```diff
@@ -398,6 +398,136 @@ func Run(o *option.Option) {
 				verifyMountsInfo(actualMount, expectedMount)
 			})
 		})
+
+		// Cgroup version v2
+		ginkgo.When("running a container with resource flags", func() {
```

Comment:
Do we want to add containerd into the go.mod or just copy the CGMode function from containerd with reference to it?

Reply:
> add containerd into the go.mod

SGTM

Reply:
> Could we change the function signature to accept RunOption instead? We can then check [CGMode](https://github.com/containerd/cgroups/blob/e8802a182774b8e25cc65b5c374389338cc8e6c3/utils.go#L39) to skip the tests will only pass on a cgroupv2 host.

Do you mean change the signature of Run(o *RunOption) function or or the `func()` in line 403? If latter, how to pass the value of RunOption to func()? Cloud you explain a bit? Thanks!

Reply:
Couldn't build cgroup on Mac OS because it's targeting at linux based systems. https://github.com/containers/podman/issues/1840

Reply:
I think only duplicating the struct doesn't work. We need to call the [Mode() ](https://github.com/containerd/cgroups/blob/e8802a182774b8e25cc65b5c374389338cc8e6c3/utils.go#L53)function to check the system cgroup version but it can only build on Linux systems. We could add a TODO for this if we want to support MacOS.
```
make run                               
go test -timeout 30m ./run/... -test.v -ginkgo.v -args --subject="finch"
# github.com/containerd/cgroups
../../../../go/pkg/mod/github.com/containerd/cgroups@v1.0.4/memory.go:463:19: undefined: unix.Eventfd
../../../../go/pkg/mod/github.com/containerd/cgroups@v1.0.4/memory.go:463:35: undefined: unix.EFD_CLOEXEC
../../../../go/pkg/mod/github.com/containerd/cgroups@v1.0.4/utils.go:70:13: undefined: unix.CGROUP2_SUPER_MAGIC
../../../../go/pkg/mod/github.com/containerd/cgroups@v1.0.4/utils.go:77:23: undefined: unix.CGROUP2_SUPER_MAGIC
FAIL    github.com/runfinch/common-tests/run [build failed]
FAIL
```

Reply:
> We need to call the [Mode()](https://github.com/containerd/cgroups/blob/e8802a182774b8e25cc65b5c374389338cc8e6c3/utils.go#L53)

Caller of this function will manually specify the cgroup version:

```go
type RunOption struct {
	BaseOpt *option.Option
	CGMode cgroups.CGMode
}
```